### PR TITLE
[Proposal] Preferences Dialog + Controlling Feature Flags

### PR DESCRIPTION
**Pull request by afiune**
## Description
Wouldn't it be great if we could turn Feature Flags on and off from the Chef Workstation App?
**We all love UI's!**

tenor-136549940

This pull request is a proposal of how we could present and introduce
Feature Flags to our users. It was originated from the Design Proposal
https://github.com/chef/chef-workstation/issues/609 

~The flow starts by, of course, turning on the new **"Preference Dialog"**
Feature (INCEPTION), to turn it on users will have to modify their
configuration file at `$HOME/.chef-workstation/config.toml` with the
following config:~
```toml
[features]
preferences_dialog = true
```
~__NOTE: This work is in flight, but when it is finished, the user won't have the
need to enable it manually and we will add release notes and documentation.__~

~Then, since we currently don't have any [file listener just yet](https://github.com/chef/chef-workstation-app/pull/50), a restart of
the Chef Workstation App is required.~

~After restart,~ users will see a new item in the Menu Bar:
Screen Shot 2020-02-10 at 2 10 31 AM

When clicking **"Preferences..."** the Preferences Dialog will pop-up: 
Screen Shot 2020-02-10 at 3 09 39 PM

By default, the **General** tab will be selected, users can navigate to other tabs
like the **Advanced** tab:
Screen Shot 2020-02-10 at 3 09 43 PM

In this tab, users can turn on and off features, if a user adds any feature directly
to their configuration file (`config.toml`), the Chef Workstation App won't be
able to override that behavior. (which is the current behavior we have with the
`channel` setting for automatic updates)

Users can also see the [Legal Disclosure](https://github.com/chef/chef-workstation-app/pull/125#issuecomment-573865697) by pressing the ❓ icon:  
Screen Shot 2020-02-10 at 3 09 50 PM

## TODO

- [X] Update the Feature Flag message with real words.
- [x] Verify with the UX about the proposed design. https://github.com/chef/chef-workstation-app/issues/129

## Related Issue

https://github.com/chef/chef-workstation/issues/609
Closes https://github.com/chef/chef-workstation/issues/724
And maybe https://github.com/chef/chef-workstation/issues/709

Split work into:
https://github.com/chef/chef-workstation-app/issues/158
https://github.com/chef/chef-workstation-app/issues/159

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.
- [x] I have run the pre-merge tests locally and they pass.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

**Comment by tyler-ball** [comment]
Lets hear from UI before we merge this one. I know we had some style guides in the past for Windows styling, so I want to make sure we can reflect that (and possibly unix styling) before we get this merged

File: src/preferences/preferences.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  // remove the selected class and add it to the new tab
+  selectedDiv.classList.remove("selected");
+  document.getElementById(tab).classList.add("selected");
+
+  // capitalize the first letter and update the title of the dialog
+  const upper = tab.charAt(0).toUpperCase() + tab.substring(1);
+  document.getElementById("window-title").textContent = upper;
+}
+
+// TODO @afiune implement this functionality
```
Comment (afiune) [inline src/preferences/preferences.ts]: IMO all these three TODOs should be done on a separate PR, if we want to use this PR I would be ok with it, just a heads-up that the PR could get a bit big since it is already `+1k` lines of code.

**Comment by tyler-ball** [comment]
Another enhancement I think we should make, and would like to get UI input - lets add descriptions for each of these feature flags. I also would like to see these options in-line in the preferences pane rather than in a separate window.

**Comment by afiune** [comment]
@tyler-ball I like the idea and we could definitely implement it, to share a little bit the "why is this a new dialog and with that specific style". The main reason is that, if we have a number of features, they will probably not fit inside the preferences dialog, by having a new dialog with a "frame that can scroll up or down" we ensure that we can add as many features as we want.

Let us wait for UX to weight in and give their recommendations. Thank you so much for your review!

**Comment by afiune** [comment]
Updated Preferences Dialog from:
Screen Shot 2019-11-13 at 10 59 18 AM

To:
Screen Shot 2019-12-05 at 2 33 16 PM

**Comment by jyan-chef** [comment]
Please using the following verbiage for the legal disclosure. 

"The warranties and indemnities in your license agreement do not apply to experimental features. Experimental features are provided "as is" with no warranties of any kind, express or implied.  Chef disclaims any and all liability related in any way to the experimental features or any related documentation."

**Comment by tyler-ball** [comment]
Moving this back to the parking lot because we do not currently have UX resources for https://app.zenhub.com/workspace/o/chef/chef-workstation-app/issues/129

**Comment by afiune** [comment]
Reduced the complexity of this PR by hiding the notifications tab and the telemetry checkbox:
https://github.com/chef/chef-workstation-app/issues/158
https://github.com/chef/chef-workstation-app/issues/159

**Comment by afiune** [comment]
@jonsmorrow yeah I know, this work has been in the works since November so it got pretty
heavy and reworked multiple times. Im glad we are getting this in for users to start using
the preferences settings at least for Mac users. Gotta tackle the Windows side of things!

tenor-51622008
